src/test/java/org/broadinstitute/hellbender/tools/exome/AllelicCNVIntegrationTest.java

```diff
@@ -63,6 +101,64 @@ public void testACNVWithAllelicPON() {
+                "--" + AllelicCNV.MAX_NUM_SIMILAR_SEGMENT_MERGING_ITERATIONS_LONG_NAME, "0",
+                "--verbosity", "INFO",
+        };
+        testACNV(arguments, outputPrefix);
+
+        //check that no similar segments were merged
+        final File initialSimilarSegmentsFile = new File(outputPrefix + "-" + AllelicCNV.INITIAL_SEG_FILE_TAG + ".seg");
+        final File finalSimilarSegmentsFile = new File(outputPrefix + "-" + AllelicCNV.FINAL_SEG_FILE_TAG + ".seg");
+        final List<ACNVModeledSegment> initialACNVModeledSegments = SegmentUtils.readACNVModeledSegmentFile(initialSimilarSegmentsFile);
+        final List<ACNVModeledSegment> finalACNVModeledSegments = SegmentUtils.readACNVModeledSegmentFile(finalSimilarSegmentsFile);
+        Assert.assertEquals(initialACNVModeledSegments.size(), finalACNVModeledSegments.size());
+    }
+
+    @Test
+    public void testACNVWithSomeSimilarSegmentMergingRefitting() {
+        final File tempDir = createTempDir("acnv-integration-with-some-sim-seg-refitting" + SAMPLE_NAME);
+        final String tempDirPath = tempDir.getAbsolutePath();
+        final String outputPrefix = tempDirPath + "/" + SAMPLE_NAME;
+
+        final String[] arguments = {
+                "--" + ExomeStandardArgumentDefinitions.TUMOR_ALLELIC_COUNTS_FILE_LONG_NAME, TUMOR_ALLELIC_COUNTS_FILE.getAbsolutePath(),
+                "--" + ExomeStandardArgumentDefinitions.TANGENT_NORMALIZED_COUNTS_FILE_LONG_NAME, COVERAGES_FILE.getAbsolutePath(),
+                "--" + ExomeStandardArgumentDefinitions.SEGMENT_FILE_LONG_NAME, SEGMENT_FILE.getAbsolutePath(),
+                "--" + AllelicCNV.OUTPUT_PREFIX_LONG_NAME, outputPrefix,
+                "--" + AllelicCNV.USE_ALL_COPY_RATIO_SEGMENTS_LONG_NAME, "true",
+                "--" + AllelicCNV.NUM_SAMPLES_COPY_RATIO_LONG_NAME, "25",
+                "--" + AllelicCNV.NUM_BURN_IN_COPY_RATIO_LONG_NAME, "10",
+                "--" + AllelicCNV.NUM_SAMPLES_ALLELE_FRACTION_LONG_NAME, "25",
+                "--" + AllelicCNV.NUM_BURN_IN_ALLELE_FRACTION_LONG_NAME, "10",
+                "--" + AllelicCNV.NUM_SIMILAR_SEGMENT_MERGING_ITERATIONS_PER_FIT_LONG_NAME, "2",
+                "--verbosity", "INFO",
+        };
+        testACNV(arguments, outputPrefix);
+
+        //check that some similar segments were merged
+        final File initialSimilarSegmentsFile = new File(outputPrefix + "-" + AllelicCNV.INITIAL_SEG_FILE_TAG + ".seg");
+        final File finalSimilarSegmentsFile = new File(outputPrefix + "-" + AllelicCNV.FINAL_SEG_FILE_TAG + ".seg");
+        final List<ACNVModeledSegment> initialACNVModeledSegments = SegmentUtils.readACNVModeledSegmentFile(initialSimilarSegmentsFile);
+        final List<ACNVModeledSegment> finalACNVModeledSegments = SegmentUtils.readACNVModeledSegmentFile(finalSimilarSegmentsFile);
+        Assert.assertTrue(initialACNVModeledSegments.size() > finalACNVModeledSegments.size());
```

Comment:
Please write a warning note in the header of `AllelicCNVIntegrationTest` mentioning that the behavior of the tests depend on the _content_ of the input files and changing them may break the tests.